Indicators/CorrelationIndicator.cs

```diff
@@ -0,0 +1,247 @@
+        /// As this indicator is receiving data points from two different symbols,
+        /// it's going to compute the next value when the amount of data points
+        /// of each of them is the same. Otherwise, it will return the last correlation
+        /// value computed
+        /// </summary>
+        /// <param name="input">The input value of this indicator on this time step.
+        /// It can be either from the target or the reference symbol</param>
+        /// <returns>The correlation value of the target used in relation with the reference</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            var inputSymbol = input.Symbol;
+            if (inputSymbol == _targetSymbol)
+            {
+                _targetDataPoints.Add(input.Close);
+            } 
+            else if(inputSymbol == _referenceSymbol)
+            {
+                _referenceDataPoints.Add(input.Close);
+            }
+            else
+            {
+                throw new ArgumentException("The given symbol was not target or reference symbol");
+            }
+
+            if (_targetDataPoints.Samples == _referenceDataPoints.Samples && _referenceDataPoints.Count > 1)
+            {
+                _targetReturns.Add(GetNewReturn(_targetDataPoints));
+                _referenceReturns.Add(GetNewReturn(_referenceDataPoints));
+
+                ComputeCorrelation();
+            }
+            return _correlation;
+        }
+
+        /// <summary>
+        /// Computes the returns with the new given data point and the last given data point
+        /// </summary>
+        /// <param name="rollingWindow">The collection of data points from which we want
+        /// to compute the return</param>
+        /// <returns>The returns with the new given data point</returns>
+        private static double GetNewReturn(RollingWindow<decimal> rollingWindow)
+        {
+            return (double) ((rollingWindow[0] / rollingWindow[1]) - 1);
+        }
+
+        /// <summary>
+        /// Computes the correlation value of the target in relation with the reference
+        /// using the target and reference returns
+        /// </summary>
+        private void ComputeCorrelation()
+        {
+
+            if (_targetDataPoints.Count < _period || _referenceDataPoints.Count < _period)
+            {
+                _correlation = 0;
+                return;
+            }
+            double _corr = 0;
+           
+
+            if (_correlationType == CorrelationIndicatorType.Pearson)
+            {
+                _corr = Correlation.Pearson(_targetDataPoints.Select(d=>(double)d).ToList(), _referenceDataPoints.Select(d => (double)d).ToList());
+            }
+            if (_correlationType == CorrelationIndicatorType.Spearman)
+            {
+                _corr = Correlation.Spearman(_targetDataPoints.Select(d => (double)d).ToList(), _referenceDataPoints.Select(d => (double)d).ToList());
+            }
```

Comment:
`Correlation.Pearson` and `Correlation.Spearman` take `IEnumerable`s, so you can drop the list conversion to avoid multiple enumerations. Could just use `Linq`'s `Cast<double>()`

Reply:
_corr must be double, because decimal do not support NaN value. Correlation can return NaN and it is there a risk of throwing exception.

Reply:
Agree with JA, shouldn't be doing `ToList()` for each sub enumerator, should handle required conversions. Maybe we could directly store doubles in the `_targetDataPoints/_referenceDataPoints` collections, no need to store decimals if we will only used them as double later (which is okay for this use case I thinK)